Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -212,21 +208,24 @@ JPEGImageIO::Read(void * buffer)
   // prepare to read the bulk data
   jpeg_start_decompress(&cinfo);
 
-  SizeValueType rowbytes = cinfo.output_components * cinfo.output_width;
-  auto *        tempImage = static_cast<JSAMPLE *>(buffer);
-
-  auto * row_pointers = new JSAMPROW[cinfo.output_height];
-  for (ui = 0; ui < cinfo.output_height; ++ui)
+  auto * volatile row_pointers = new JSAMPROW[cinfo.output_height];
+  auto * tempImage = static_cast<JSAMPLE *>(buffer);
+  for (size_t ui = 0; ui < cinfo.output_height; ++ui)
   {
-    row_pointers[ui] = tempImage + rowbytes * ui;
+    row_pointers[ui] = tempImage + cinfo.output_components * cinfo.output_width * ui;
   }
 
   // read the bulk data
-  unsigned int remainingRows;
   while (cinfo.output_scanline < cinfo.output_height)
   {
-    remainingRows = cinfo.output_height - cinfo.output_scanline;
-    jpeg_read_scanlines(&cinfo, &row_pointers[cinfo.output_scanline], remainingRows);
+    if (setjmp(jerr.setjmp_buffer))
+    {
+      itkWarningMacro("JPEG error in the file " << this->GetFileName());
```

Comment:
If this means that the file cannot be read, I think it should say so in the message and possibly trigger an exception. Maybe other maintainers can comment better on the warning vs. exception question.

Reply:
If I understand correctly this situation is similar to this other one for PNG files:
https://github.com/InsightSoftwareConsortium/ITK/pull/2942/files#diff-1b5f81c245a7c32cc9d0efe02aac3fd4ca831e5b3c7a5cdaa63d575821b8ad58R359

And there an exception is being added.

Reply:
It is OK. At this point warning is sufficient, the image will be returned. There are different scenarios for different broken image, depends how JPEG library handles them, some causes a warning without jumping, some with, some will be refused, it depends.
The implementation is similar with Gimp and other, i think Gimp opened billions of JPEG images on all possible platforms and is reliable.

Reply:
OK.

Reply:
In fact it is not important for me personally, for me is important to avoid crashes, they are imported into my app. If you like we can trigger exception here. But in fact, Gimp, Eye of Mate, also Google drive preview are showing thia image like we do now, probably the logic is "if possible to show partially, do it".